helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,35 +573,41 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has never exist in the cluster
+      String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);
+      String errMessage;
+      if (!_zkClient.exists(instancePath)) {
+        errMessage = String.format(
+            "Can't reset state for %s.%s on %s, because %s has never existed" + " in cluster %s",
```

Comment:
Why there is still a "+", it is not necessary, right?

Reply:
BTW, if this is for the format, just use the auto format supported by the IDE after you import the Helix code style.

Reply:
The auto formatter collapse 2 lines into 1 line and leave the "+" here.  I missed it.  Will fix